src/command_controller.rs

```diff
@@ -0,0 +1,288 @@
+    /// Retrieve the potential "Owner" name of this custom resource
+    fn get_owner_name(&self) -> String;
+}
+
+struct CommandState<T> {
+    context: ReconciliationContext<T>,
+}
+
+impl<T> CommandState<T>
+where
+    T: CommandOwner + Clone + DeserializeOwned + Meta,
+{
+    /// This controller sets the owner reference in our custom resource object. We need to
+    /// find the potential owner, extract its metadata into an OwnerReference object and
+    /// patch our command custom resource with that OwnerReference. This makes sure we can
+    /// list and work with the commands in the main controller loop.
+    async fn set_owner_reference(&mut self) -> ReconcileResult<Error> {
+        let owner = find_owner::<T::Owner>(
+            &self.context.client.clone(),
+            &self.context.resource.get_owner_name(),
+        )
+        .await?;
+
+        let owner_reference =
+            metadata::object_to_owner_reference::<T::Owner>(owner.meta().clone(), true)?;
+
+        patch_owner_reference(
+            &self.context.client,
+            &self.context.resource,
+            &owner_reference,
+        )
+        .await?;
+
+        Ok(ReconcileFunctionAction::Done)
+    }
+}
+
+impl<T> ReconciliationState for CommandState<T>
+where
+    T: CommandOwner + Clone + DeserializeOwned + Meta + Send + Sync,
+{
+    type Error = Error;
+
+    fn reconcile(
+        &mut self,
+    ) -> Pin<Box<dyn Future<Output = Result<ReconcileFunctionAction, Self::Error>> + Send + '_>>
+    {
+        Box::pin(async move { self.set_owner_reference().await })
+    }
+}
+
+#[derive(Debug)]
+struct CommandStrategy<T> {
+    // TODO: Better workaround for PhantomData?
+    // We use it here cause we need to make CommandStrategy generic to be able to do:
+    // impl<T> ControllerStrategy for CommandStrategy<T>
+    _ignore: Option<std::marker::PhantomData<T>>,
+}
+
+impl<T> CommandStrategy<T> {
+    pub fn new() -> CommandStrategy<T> {
+        CommandStrategy { _ignore: None }
+    }
+}
+
+#[async_trait]
+impl<T> ControllerStrategy for CommandStrategy<T>
+where
+    T: CommandOwner + Meta + Clone + DeserializeOwned + Send + Sync,
+{
+    type Item = T;
+    type State = CommandState<T>;
+    type Error = Error;
+
+    fn finalizer_name(&self) -> String {
+        FINALIZER_NAME.to_string()
+    }
+
+    async fn init_reconcile_state(
+        &self,
+        context: ReconciliationContext<Self::Item>,
+    ) -> Result<Self::State, Error> {
+        Ok(CommandState { context })
+    }
+}
+
+/// This creates an instance of a [`Controller`] which waits for incoming events and reconciles them.
+///
+/// This is an async method and the returned future needs to be consumed to make progress.
+pub async fn create_command_controller<T>(client: Client)
+where
+    T: CommandOwner + Meta + Clone + Debug + DeserializeOwned + Send + Sync + 'static,
+{
+    let command_api: Api<T> = client.get_all_api();
+    let owner_api: Api<T::Owner> = client.get_all_api();
```

Comment:
Help me please...why do we even need to watch the Owner?

As mentioned above this should probably just take a second generic parameter instead of using T::Owner